src/solver/optimisation/opt_optimisation_lineaire.cpp

```diff
@@ -89,13 +100,22 @@ bool OPT_OptimisationLineaire(PROBLEME_HEBDO* ProblemeHebdo, uint numSpace)
         OPT_InitialiserLesCoutsLineaire(
           ProblemeHebdo, PremierPdtDeLIntervalle, DernierPdtDeLIntervalle, numSpace);
 
-        if (!OPT_AppelDuSimplexe(ProblemeHebdo, NumeroDeLIntervalle, optimizationNumber))
+        // An optimization period represents a sequence as <year>-<week> or <year>-<week>-<day>,
+        // depending whether the optimization is daily or weekly.
+        // These sequences are used when building the names of MPS or criterion files. 
+        auto opt_period_as_string = createOptimizationPeriod(ProblemeHebdo->OptimisationAuPasHebdomadaire,
+                                                             NumeroDeLIntervalle,
+                                                             ProblemeHebdo->weekInTheYear,
+                                                             ProblemeHebdo->year);
+
+        if (!OPT_AppelDuSimplexe(ProblemeHebdo, NumeroDeLIntervalle, optimizationNumber, opt_period_as_string))
```

Comment:
optimizationRangeStringGenerator

Reply:
Not sure **range** it is better than **period**

Reply:
Done : **opt_period_as_string** ==> **optPeriodStringGenerator**